### DISCUSSION TITLE
Simple question: changing types on the way out

### DISCUSSION THREAD

**Discussion by ajoslin** [hayes/pothos#481] [Q&A] [answered]
Hi, first - LOVE pothos + prisma. We are using it extensively. ❤️ Thank you @hayes.

Question: In pothos, often we query a relation in a way that we want a boolean. For example, we have a `User` prisma atype and I want to add a field `isSelfFollowing` that checks if the logged in user making the query is following the `User` being queried.

To do this, I set up a `relation('followers')` and add `context.userId` to the `query` for that relation. This ends up returning an array of followers, which is kind of a cumbersome way for the API to express whether or not self is following the queried `User`.

Is there a way to get Pothos to create these optimal joined Prisma queries, as `relation`+`query:()=>{}` does, THEN on the way out map something like this to a boolean?

**Comment by hayes** [comment, marked as answer]
You could do something like this:

```ts
builder.prismaObject('Post', {
  fields: (t) => ({
    id: t.exposeID('id'),
    title: t.exposeString('title'),
    content: t.exposeString('content'),
    author: t.relation('author'),
    comments: t.relation('comments'),
    hasCommented: t.boolean({
      select: (args, ctx) => ({
        comments: {
          where: {
            author: {
              id: ctx.user!.id,
            },
          },
          take: 1,
          // only load the id (optional)
          select: {
            id: true,
          },
        },
      }),
      resolve: (post) => post.comments.length > 0,
    }),
  }),
});
```
